x-pack/plugins/actions/server/builtin_action_types/pagerduty.test.ts

```diff
@@ -142,6 +143,16 @@ describe('validateParams()', () => {
 - [eventAction.2]: expected value to equal [acknowledge]"
 `);
   });
+
+  test('should validate and throw error when timestamp is invalid', () => {
+    const randoDate = new Date('1963-09-23T01:23:45Z').toISOString();
+    const timestamp = `${repeat(' ', random(0, 10))}${randoDate}${repeat(' ', random(0, 10))}`;
```

Comment:
I don't think we need the randomness here.  Just a date that can't be parsed correctly.

Reply:
Since one of the issues fixed by this PR (haven't read all the way through yet) is that some date params had leading/trailing spaces, probably makes sense to test with an overtly bad data - `"bad-date-string"` as well as - like above - a date with leading/trailing spaces.

Reply:
It's an attempt at a poor man's property-based testing 🤷 
Not too fussed, I'll remove the random